pkg/lvm/lvm_util.go

```diff
@@ -86,7 +86,33 @@ func buildLVMCreateArgs(vol *apis.LVMVolume) []string {
 	}
 
 	LVMVolArg = append(LVMVolArg, vol.Spec.VolGroup)
+	return LVMVolArg
+}
+
+// builldThinLVMCreateArgs returns lvcreate command for thin volume
+func buildThinLVMCreateArgs(vol *apis.LVMVolume) []string {
+	var LVMVolArg []string
+
+	volume := vol.Name
+	size := vol.Spec.Capacity + "b"
+	pool := vol.Spec.VolGroup + "_thinpool"
+
+	if len(vol.Spec.Capacity) != 0 {
+		// check if thin pool exists for given volumegroup
+		if !lvThinExists(vol.Spec.VolGroup, pool) {
```

Comment:
why this check? what will happen if thin pool is there?

Reply:
if the 
- thin pool exists for given volumegroup: create command will not have pool args( -L flag hrere) hence the pool will not be created. (single thin pool can be used for multiple thin volumes via storageclass)
- thin pool not exists for given volumegroup: create command will have pool args  included,which will create the thin pool as well with thin volume.